pkg/downloader/chart_downloader.go

```diff
@@ -97,6 +98,13 @@ func atomicWriteFile(filename string, body io.Reader, mode os.FileMode) error {
 	return os.Rename(tempName, filename)
 }
 
+type InMemoryChart struct {
+	Chart                  *bytes.Buffer
+	ProvenanceVerification *provenance.Verification
+}
+
+var InMemoryCache = make(map[string]InMemoryChart)
+
```

Comment:
Can you explain why we need a map? Are there use-cases where we need to store more than one chart at a time?

Reply:
While installing charts with dependencies multiple charts would be downladded and stored. Let me know if I am missing a fact here.

Reply:
I think we don't really need it as of now, but for the sake of semantics we should keep it. Otherwise what is tracking the InMemory Charts ?

Reply:
This is important as this would be used by `cmd` to query the charts.

Reply:
I guess my question was: do we need to put more than one chart at a time in the InMemoryCache, or do we only need one chart.  If we only need one chart, the InMemoryCache could simply be `var InMemoryCache InMemoryChart`